static/js/apps/eval_rig/app.tsx

```diff
@@ -17,25 +17,83 @@
 import { OAuthCredential, User } from "firebase/auth";
 import { GoogleSpreadsheet } from "google-spreadsheet";
 import React, { useState } from "react";
-import { Col, Row } from "reactstrap";
 
 import { GoogleSignIn } from "../../utils/google_signin";
+import { Query, QuerySection } from "./query_section";
 
-export function App(): JSX.Element {
+interface AppPropType {
+  sheetId: string;
+}
+
+export function App(props: AppPropType): JSX.Element {
   const [user, setUser] = useState<User | null>(null);
+  const [doc, setDoc] = useState<GoogleSpreadsheet>();
+  const [allQuery, setAllQuery] = useState<Record<string, Query>>({});
+  const [allCall, setAllCall] = useState<
+    Record<string, Record<string, number>>
+  >({});
+
+  const loadQuery = (doc: GoogleSpreadsheet) => {
+    doc.loadInfo().then(() => {
+      const qaSheet = doc.sheetsByTitle["query_and_answer"];
+      const qaRows = qaSheet.rowCount;
+      qaSheet
+        .loadCells({
+          startRowIndex: 1,
+          endRowIndex: qaRows,
+          startColumnIndex: 0,
+          endColumnIndex: 3,
+        })
+        .then(() => {
+          const allQuery: Record<string, Query> = {};
+          for (let i = 1; i < qaRows; i++) {
```

Comment:
can you continue when `qaSheet.getCell(i, 0)` is empty?

Reply:
Yes, ok for now, but I feel it might be useful to be resilient to a stale empty row.  We can see how it plays out...